### DISCUSSION TITLE
Per-World Datapacks 2

### DISCUSSION THREAD

**Discussion by Etanarvazac** [PaperMC/Paper#10518] [Ideas]
### Is your feature request related to a problem?

This was originally mentioned by **Prototype464** in issue #7347. It's been **_2 years_** since their inquiry into this. As they have stated, it would be great to make some datapacks per-world. For example, applies something like Terralith to a resource-only world. I will note the end goal here is well-put by **Prototype464** in their issue.

### Describe the solution you'd like.

The ability to use datapacks per-world on a server. For example, Terralith limited to a resource-only world.

### Describe alternatives you've considered.

Like **Prototype464**, I have been unsuccessful in finding a plugin for this.

**Comment by Machine-Maker** [comment]
This is slightly more feasible with the introduction of the configuration stage. That essentially means the player leaves the server and re-joins, but to the client it is just a loading screen.

No such API exists for putting a client back in that configuration phase as of yet and it can't be added very easily without reworking a lot of the changes craftbukkit has made over the years to how players are created. It's possible, but not tomorrow and probably not the next day.

EDIT: You can accomplish this essentially with a network connected via a proxy like Velocity.

**Comment by qyl27** [comment]
How about the datapacks with only the world generation part?  
Recipes and advancements may should be applied globally, but world and structures generation needn't.

**Comment by electronicboy** [comment]
It’s still global state, you would need to rename the entries inside of the
data pack to apply to the dimension you want, this is generally not
something that we can support, especially given that it would require out
of phase modifications of global state, or further mangling of the registry
system. If you want to use data packs to modify a custom dimension, you’d
probably be better off defining that dimension in the pack rather than with
a plugin

**Reply by qyl27** [reply]
ah, thank you
